File: video/list-videos-direct-pagination.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 
  const listParams = {
    id: "",
    filename: "",
    size: 1024,
    width: 1280,
    height: 720,  // Example: 720p to 1080p
    duration: 20,  // Example: 1 minute to 5 minutes
    fps: 24,
    metadata: {
      category: "nature"
    },
    createdAt: "2024-09-17T07:53:46.365Z",
    updatedAt: "2024-09-17T07:53:46.365Z",
    indexedAt: "2024-09-17T07:55:22.125Z",
    page: 1,
    pageLimit: 5,
    sortBy: "created_at",
    sortOption: "desc"
  };
  
  const videos = await client.index.video.list("", listParams)
  videos.forEach(video => {
    console.log(`ID: ${video.id}`);
    console.log(`  Created at: ${video.createdAt}`);
    console.log(`  Updated at: ${video.updatedAt || "N/A"}`);
    console.log(`  Indexed at: ${video.indexedAt || "N/A"}`);
    console.log("  Metadata:");
    console.log(`    Filename: ${video.metadata.filename}`);
    console.log(`    Duration: ${video.metadata.duration}`);
    console.log(`    FPS: ${video.metadata.fps}`);
    console.log(`    Width: ${video.metadata.width}`);
    console.log(`    Height: ${video.metadata.height}`);
    console.log(`    Size: ${video.metadata.size}`);
    if (video.hls) {
      console.log("  HLS:");
      console.log(`    Video URL: ${video.hls.videoUrl || "N/A"}`);
      console.log("    Thumbnail URLs:");
      (video.hls.thumbnailUrls || []).forEach((url) => {
        console.log(`      ${url}`);
      });
      console.log(`    Status: ${video.hls.status || "N/A"}`);
      console.log(`    Updated At: ${video.hls.updatedAt}`);
    }
    if (video.source) {
      console.log("  Source:");
      console.log(`    Type: ${video.source.type}`);
      console.log(`    Name: ${video.source.name}`);
      console.log(`    URL: ${video.source.url || "N/A"}`);
    }
  });
})();

File: video/list-videos-direct-pagination.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

videos = client.index.video.list(
    index_id="",
    id="",
    filename="01.mp4",
    size=1024,
    width=920,
    height=1080,
    duration=100,
    fps=30,
    metadata={"category": "nature"},
    created_at="2024-09-17T07:53:46.365Z",
    updated_at="2024-09-17T07:53:46.365Z",
    indexed_at="2024-09-17T07:55:22.125Z",
    page=1,
    page_limit=5,
    sort_by="created_at",
    sort_option="desc"
)
for video in videos:
    print(f"ID: {video.id}")
    print(f"  Created at: {video.created_at}")
    print(f"  Updated at: {video.updated_at}")
    print(f"  Indexed at: {video.indexed_at}")
    print("  Metadata:")
    print(f"    Filename: {video.metadata.filename}")
    print(f"    Duration: {video.metadata.duration}")
    print(f"    FPS: {video.metadata.fps}")
    print(f"    Width: {video.metadata.width}")
    print(f"    Height: {video.metadata.height}")
    print(f"    Size: {video.metadata.size}")
    if video.hls:
        print("  HLS:")
        print(f"    Video URL: {video.hls.video_url}")
        print("    Thumbnail URLs:")
        for url in video.hls.thumbnail_urls or []:
            print(f"      {url}")
        print(f"    Status: {video.hls.status}")
        print(f"    Updated At: {video.hls.updated_at}")
    if video.source:
        print("  Source:")
        print(f"    Type: {video.source.type}")
        print(f"    Name: {video.source.name}")
        print(f"    URL: {video.source.url}")

